proposals/NNNN-byte-address-buffer-alignment.md

```diff
@@ -0,0 +1,983 @@
+---
+title: "NNNN - ByteAddressBuffer Alignment
+params:
+  authors:
+     - mapodaca-nv: Mike Apodaca
+   sponsors:
+     - tex3D: Tex Riddell
+   status: Under Consideration
+---
+
+* Required Version: Shader Model X.Y, Vulkan X.Y, and/or HLSL 20XY
+* Issues: [#543](https://github.com/microsoft/hlsl-specs/issues/543),
+   [#258](https://github.com/microsoft/hlsl-specs/issues/258)
+
+## Introduction
+
+This proposal introduces a `BaseAlignment` attribute for byte address buffer object declarations that specifies base
```

Comment:
I wonder if we should use a different name for this attribute.

`BaseAlignment` sounds pretty generic, while this is meant to apply specifically to a buffer resource view. Intuitively it sounds like it could apply to other things, like global/static/groupshared variable alignment, or as HLSL gains features, things like pointers or references.

How about `BufferAlignment`? This indicates that the buffer resource (view) needs to be aligned in this way. It makes it clear that it applies to the buffer, rather than an HLSL variable alignment.

We could expand support to `[RW]StructuredBuffer` and even `[RW]Buffer` (for certain pass-through `uint4/mixed` type scenarios, like that used for tbuffers), where "Buffer" still makes sense.